R/obj_namespace_name.R

```diff
@@ -3,6 +3,16 @@ env_ns_name <- function(e) {
   unname(getNamespaceName(e))
 }
 
+env_name <- function(e) {
+  # NOTE:
+  #   If this ever becomes problematic, we can embed C call to grab environment
+  #   pointer address instead of using output. Plenty of prior art, see
+  #   envnames::address or data.table::address for examples.
+
+  # because environment may have attributes, only take first line of output
+  sub("<environment: (.*)>", "\\1", capture.output(e)[[1L]])
+}
```

Comment:
I'm not thrilled about parsing print output, but I also don't want to make the jump to embedding C code just for fetching a pointer. It's possible that some crazy environment object in some package has tons of attributes to the point that collecting output is problematic, but I think that's very unlikely.

If we do ever want to do this, it seems to be a solved problem and [there are packages out there that implement this already](https://github.com/mastropi/envnames/blob/master/src/address.c). I'm not sure it's worth taking either of the mentioned packages as a dependency, but we could certainly adapt the small bit of code needed to fetch an environment pointer.

Reply:
I understand your concerns, but I also agree that it is really unlikely to cause any problem. Going down to C just to resolve a point would be a big overhead that in the current situation seems unnecessary. Considering we already have a plan B if this doesn't work I'm in favor of merging this as it is.